util/json.go

```diff
@@ -115,15 +118,8 @@ func Unmarshal(bs []byte, v interface{}) error {
 		return UnmarshalJSON(nbs, v)
 	}
 	// not json or yaml: try extensions
-	if value, ok := v.(*any); ok {
-		if handler := extension.FindExtension(".json"); handler != nil {
-			retval, err := handler(bs)
-			if err != nil {
-				return err
-			}
-			*value = retval
-			return nil
-		}
+	if handler := extension.FindExtension(".json"); handler != nil {
```

Comment:
We're giving the handler the control over what to attempt to handle. If it wants to handle more than just `*any`, let it.